cmd/snap-update-ns/main.go

```diff
@@ -77,7 +77,7 @@ func run() error {
 		return fmt.Errorf("cannot open lock file for mount namespace of snap %q: %s", snapName, err)
 	}
 	defer lock.Close()
-	if err := lock.Lock(); err != nil {
+	if err := lock.Lock(0); err != nil {
```

Comment:
I was also thinking about a separate call instead of the flags. My main blocker was that I did not find a good pair of names :P - I'm not super keen on `TryLock` as (to me at least) it does not convey the "try-to-lock-and-keep-the-lock-on-success. So maybe (LockBlocking, Lock) or (Lock, LockNonBlocking), (LockBlocking, LockNonblock). Better ideas welcome!